SHAPETEST/StartShapeTestLosProbe.md

```diff
@@ -23,8 +23,8 @@ enum TraceFlags
   IntersectWater = 32,
   Unknown = 128,
   IntersectFoliage = 256,
-  Unkown = 511, (used most)
-  IntersectEverything = 4294967295
+  IntersectEverything = 511
+  IntersectEverything = -1
```

Comment:
Why would you define `IntersectEverything` twice, that's wrong 🙂

Reply:
Aren't they the same basically?

Reply:
> Aren't they the same basically?

No, that's wrong. Flags are usually represented using powers of two. Each flag is assigned a unique value that is a power of two (1, 2, 4, 8, 16, etc...), and we combine these values using bitwise operations to represent different sets of options.

Now, the part that doesn't make sense is defining 511 (for `IntersectEverything`) as a flag in the enum. In proper flag representation, we should stick to powers of two. 511 is not a power of two; it's `2^9 - 1`. That's why you see it displayed as `4294967295 (0xFFFFFFFF)` also **(known as -1, which is the correct representation here)**, which is the maximum unsigned value for a 32-bit integer.

To avoid confusion and ensure consistency, it's best to correct the enum by using `-1` for `IntersectEverything`, as that's a common convention to represent all available options. This way, the enum will work correctly with bitwise operations, and we won't encounter display errors or unexpected behavior.

In other words, remove `IntersectEverything = 511` from the enum.
More about power of two [here](https://en.wikipedia.org/wiki/Power_of_two) (Computer Science section) and [here](https://en.wikipedia.org/wiki/Bitwise_operation) for bit-wise operations.

Essentially, `TraceFlags` should be this:

```c
enum TraceFlags
{
    None = 0,
    IntersectWorld = 1,                 // 2^0
    IntersectVehicles = 2,              // 2^1
    IntersectPedsSimpleCollision = 4,   // 2^2
    IntersectPeds = 8,                  // 2^3
    IntersectObjects = 16,              // 2^4
    IntersectWater = 32,                // 2^5
    Unknown = 64,                       // 2^6
    IntersectFoliage = 128,             // 2^7
    IntersectEverything = -1,           // All flags combined using bitwise OR (Power of 2 combination)
}
```

Reply:
For reference, `0x140D7CF2B/2904` maps these intersection flags to [composite flags](https://github.com/dexyfex/CodeWalker/blob/master/CodeWalker.Core/GameFiles/Resources/Bounds.cs#L4239)[^1] 

[^1]: Well, what is labelled in my IDB as `EBoundCompositeFlags` anyway.

Reply:
`511` is still a valid flag. Probably remove it but make a note that this flag exist.